scene/resources/material.cpp

```diff
@@ -886,9 +886,11 @@ void BaseMaterial3D::_update_shader() {
 	// Generate list of uniforms.
 	code += vformat(R"(
 uniform vec4 albedo : source_color;
-uniform sampler2D texture_albedo : source_color, %s;
-)",
-			texfilter_str);
+)");
+
+	if (true || textures[TEXTURE_ALBEDO].is_valid() || (flags[FLAG_ALBEDO_TEXTURE_MSDF] && !flags[FLAG_UV1_USE_TRIPLANAR])) {
```

Comment:
The `true ||` is here to work around an issue with Label3D rendering (which uses `FLAG_ALBEDO_TEXTURE_MSDF`. If I don't add it, Label3D rendering breaks as seen in the 3D platformer demo:

![image](https://github.com/user-attachments/assets/35558803-838c-4e20-844c-43f56ec2d41e)

It should look like this instead:

![image](https://github.com/user-attachments/assets/ae75b1cf-65de-481c-afdf-c824456981a4)

Reply:
I think this is happening because you aren't generating a shader variant for this condition. Everything that changes how the shader is generated needs to have a key. 

For reference, see the process of adding this optimization in 3.x:
original PR: https://github.com/godotengine/godot/pull/33097
Workaround for Sprite3D (same as Label3D): https://github.com/godotengine/godot/pull/33277
Material keys: https://github.com/godotengine/godot/pull/33683

Reply:
I see, I've left the albedo texture sample in place (most materials in production will use it anyway).